unison-core/src/Unison/Term.hs

```diff
@@ -851,7 +851,12 @@ generalizedDependencies
 generalizedDependencies termRef typeRef literalType dataConstructor dataType effectConstructor effectType
   = Set.fromList . Writer.execWriter . ABT.visit' f where
   f t@(Ref r) = Writer.tell [termRef r] $> t
-  f t@(TermLink r) = Writer.tell [termRef $ Referent.toReference r] $> t
+  f t@(TermLink (Referent.Ref r)) =
+    Writer.tell [termRef r] $> t
+  f t@(TermLink (Referent.Con r id CT.Data))
+    = Writer.tell [dataConstructor r id] $> t
+  f t@(TermLink (Referent.Con r id CT.Effect))
+    = Writer.tell [effectConstructor r id] $> t
```

Comment:
Maybe more readable if we  just have one equation and then `case r of` to get the RHSs onto the RHS, so to speak.